MapboxCoreNavigation/RouteController.swift

```diff
@@ -236,20 +236,30 @@ extension RouteController: CLLocationManagerDelegate {
             if !isCloseToCurrentStep && isCloseToUpComingStep {
                 
                 // Increment the step
-                routeProgress.currentLegProgress.stepIndex += 1
```

Comment:
The next line depends on `routeProgress.currentLegProgress.currentStep`. Now it’ll be snapping the distance to a different maneuver than before.